eta/core/data.py

```diff
@@ -1029,3 +1031,11 @@ def optional(cls):
     @classmethod
     def required(cls):
         return ["video_path", "label"]
+
+
+class FeatureArray(NpzWriteable):
```

Comment:
The semantics of FeatureArray are not exact enough for me.  `y` is not features.   `X` could plausibly be raw data.  Does this name need to change?

Reply:
@jasoncorso is this just a question of the name of the class, or somehow clarifying what `y` and `X` actually are?  For the record, @brimoor 's description is accurate:
```
X is an n x d array whose rows contain features
y is a length-n array of labels
```

Reply:
The goal is just to make the name + documentation more descriptive of what we actually intend to store in this class. I'm fine with keeping the scope of this type to a collection of labeled feature vectors (which could be vectorized raw images, if desired)